src/core/a-scene.js

```diff
@@ -402,13 +402,25 @@ var AScene = module.exports = registerElement('a-scene', {
 
     setupCanvas: {
       value: function () {
-        var canvas = this.canvas = document.createElement('canvas');
+        var canvasSelector = this.getAttribute('canvas');
+        var canvas;
+
+        if (canvasSelector) {
+          canvas = this.canvas = document.querySelector(canvasSelector);
```

Comment:
If the selector is invalid (an empty string for instance) a DOMException will raise. Should we capture the exception?